dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java

```diff
@@ -166,4 +178,60 @@ public Response renderPage(@Context final HttpServletRequest request, @Context f
         return res;
     }
 
+    @NoCache
+    @POST
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    @Path("/{pageId}/layout")
+    public Response saveLayout(@Context final HttpServletRequest request, @PathParam("pageId") final String pageId,
+                               final String json) {
+
+        final InitDataObject auth = webResource.init(false, request, true);
+        final User user = auth.getUser();
+
+        Response res = null;
+
+        try {
+
+            Template templateSaved = this.pageResourceHelper.saveTemplate(user, pageId, json);
+            res = Response.ok(new ResponseEntityView(templateSaved)).build();
+
+        } catch (DotSecurityException e) {
+            res = ExceptionMapperUtil.createResponse(e, Response.Status.UNAUTHORIZED);
+        } catch(NotFoundException e) {
+            res = ExceptionMapperUtil.createResponse(e, Response.Status.NOT_FOUND);
+        } catch (BadRequestException | DotDataException e) {
+            res = ExceptionMapperUtil.createResponse(e, Response.Status.BAD_REQUEST);
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)